models/deepseek/v4/prefill_attention_csa.py

```
@@ -444,277 +477,424 @@ def golden_prefill_attention_csa(tensors):
+            abs_pos = int(pos[t].item())
+            window_valid = min(WIN, abs_pos + 1)
+            key_start_abs = abs_pos + 1 - window_valid
+            for k, key_abs in enumerate(range(key_start_abs, abs_pos + 1)):
+                overlay_t = current_by_req[req].get(key_abs)
+                if overlay_t is not None and overlay_t <= t:
+                    topk_idxs[t, k] = WIN + overlay_t
+                else:
+                    topk_idxs[t, k] = key_abs % WIN
+            cursor = window_valid
+            visible_cmp = (abs_pos + 1) // COMPRESS_RATIO
+            for ck, raw_t in enumerate(cmp_topk[t, :compressed_cap].tolist()):
```

Ruff (0.15.15)

[warning] Loop control variable `ck` not used within loop body

Rename unused `ck` to `_ck`

(B007)